### ISSUE TITLE
How to `run_now()` `with_temp_loop()`?

### ISSUE THREAD

**Issue by mmuurr** [r-lib/later#165]
I'm probably missing something very obvious here, but I'm also completed stumped as to how to use `with_temp_loop()`.
Here's a trivial global-loop example:
```R
later::later(function() print("foo"), delay = 2)
```
Works great; prints "foo" after a few seconds.

Now I wanted to pop this onto a different (i.e. non-global) loop ... one way is to formally create a loop and use that loop's reference. Per the documentation, for a one-time-only later execution we can also do something like this(?):
```R
later::with_temp_loop(
  later::later(function() print("foo"), delay = 2)
)
```
But this never seems to execute. And since temp loops are created with [`parent=NULL`](https://github.com/r-lib/later/blob/main/R/later.R#L164) (and thus the global loop's `run_now()` doesn't trigger this temp loop to run), it's not obvious to me how to force execution with `run_now()` (since we don't have a reference to the loop).

Perhaps related to rstudio/httpuv#250?

**Comment by mmuurr** [comment]
@wch aaaahhhh that makes sense, I  I also assume, then, that something like this (swapping the delay & timeout values in your example) would be an anti-pattern preventing execution:
```R
with_temp_loop({
  later(function() print("foo"), delay = 2)
  run_now(1)  ## too soon, as the `later` function isn't ready yet.
})
```
If I gin up some examples (using some of your examples from this thread), would it be worth me adding to the documentation and submitting a PR (or do y'all prefer to write your own documentation internally)?
In either case, thanks for the thoughtful reply, this helped me a lot!

**Comment by wch** [comment]
Yes, thanks, that would be good to add in the documentation! If you are feeling ambitious, a new page about private event loops would be a great thing to have.
